src/components/vm/consoles/serial.tsx

```diff
@@ -60,22 +74,20 @@ export class SerialActive extends React.Component {
         };
 
         if (channel_needs_update())
-            this.updateChannel(this.props.spawnArgs);
+            this.updateChannel();
     }
 
     updateChannel() {
         if (this.state.channel)
             this.state.channel.close();
 
-        if (this.props.state.connected) {
-            const opts = {
+        if (this.props.state.connected && this.props.spawnArgs.length > 0) {
+            const channel = cockpit.channel({
                 payload: "stream",
                 spawn: this.props.spawnArgs,
                 pty: true,
-            };
-            if (this.props.connectionName == "system")
-                opts.superuser = "try";
-            const channel = cockpit.channel(opts);
+                ...(this.props.connectionName == "system" ? { superuser: "try" } : { })
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2267-42a6eadf-20250808-101729-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)